tool/ruby_vm/scripts/insns2vm.rb

```diff
@@ -0,0 +1,88 @@
+#! /your/favourite/path/to/ruby
+# -*- mode: ruby; coding: utf-8; indent-tabs-mode: nil; ruby-indent-level: 2 -*-
+# -*- frozen_string_literal: true; -*-
+# -*- warn_indent: true; -*-
+#
+# Copyright (c) 2017 Urabe, Shyouhei.  All rights reserved.
+#
+# This file is  a part of the programming language  Ruby.  Permission is hereby
+# granted, to  either redistribute and/or  modify this file, provided  that the
+# conditions  mentioned in  the file  COPYING are  met.  Consult  the file  for
+# details.
+
+require 'optparse'
+require_relative '../controllers/application_controller.rb'
+
+def router argv
```

Comment:
Minor: Although I understand it's normal in one-shot Ruby scripts, I'm softly against defining method at top-level even in non-production application because:

* It makes harder to find definition by grep, especially in the case that the caller is in another file
  * `router` wasn't found in another place for now but it may be added in the future as it's a common word
  * My suggestion is changing this to a singleton method of module like `Insns2vm.router`
* `AnyClass.router` can be called, which sounds unintentional
  * Not blocking issue: And it may break some code that depends on method_missing